### PR TITLE
[opt](profile) add index page profile for io

### PR DESCRIPTION
**Pull request by airborne12**
### What problem does this PR solve?

Related PR: #42697

Problem Summary:

Add zonemap/primary key/bloom filter index page stats into profile

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
lgtm

File: be/src/olap/rowset/segment_v2/bloom_filter_index_reader.cpp [hunk 1]
```diff
@@ -45,20 +44,21 @@ int64_t BloomFilterIndexReader::get_metadata_size() const {
...
 }
 
-Status BloomFilterIndexReader::_load(bool use_page_cache, bool kept_in_memory) {
+Status BloomFilterIndexReader::_load(bool use_page_cache, bool kept_in_memory,
+                                     OlapReaderStatistics* index_load_stats) {
     const IndexedColumnMetaPB& bf_index_meta = _bloom_filter_index_meta->bloom_filter();
 
     _bloom_filter_reader.reset(new IndexedColumnReader(_file_reader, bf_index_meta));
-    RETURN_IF_ERROR(_bloom_filter_reader->load(use_page_cache, kept_in_memory, _index_load_stats));
-    update_metadata_size();
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/bloom_filter_index_reader.cpp:53]: why delete this line?

File: be/src/olap/rowset/segment_v2/bloom_filter_index_reader.h [hunk 2]
```diff
@@ -48,17 +48,18 @@ class BloomFilterIndexReader : public MetadataAdder<BloomFilterIndexReader> {
...
     Status load(bool use_page_cache, bool kept_in_memory,
-                OlapReaderStatistics* _bf_index_load_stats = nullptr);
+                OlapReaderStatistics* bf_index_load_stats);
 
     BloomFilterAlgorithmPB algorithm() { return _bloom_filter_index_meta->algorithm(); }
 
     // create a new column iterator.
-    Status new_iterator(std::unique_ptr<BloomFilterIndexIterator>* iterator);
+    Status new_iterator(std::unique_ptr<BloomFilterIndexIterator>* iterator,
+                        OlapReaderStatistics* index_load_stats = nullptr);
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/bloom_filter_index_reader.h:57]: why there is a still defaul value ?

File: be/src/olap/rowset/segment_v2/indexed_column_reader.h [hunk 3]
```diff
@@ -103,7 +104,6 @@ class IndexedColumnReader : public MetadataAdder<IndexedColumnReader> {
     const KeyCoder* _value_key_coder = nullptr;
     uint64_t _mem_size = 0;
     bool _is_pk_index = false;
-    OlapReaderStatistics* _index_load_stats = nullptr;
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/indexed_column_reader.h:106]: also should modify IndexedColumnIterator, it also saves OlapReaderStatistics* _stats = nullptr; at line 192

**Comment by airborne12** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**